### ISSUE TITLE
Hidden unit attack of opportunity deadlock when owning player not connected

### ISSUE THREAD

**Issue by mjog** [MegaMek/megamek#3045]
### Environment

Megamek 0.48.0
Clients: Windows Oracle Java 8 & Linux OpenJDK 11
Dedicated server: Linux OpenJDK 8 VM on a Linode-like web host
Game setup: two players, ~100 units each, 2x2 64x64 maps, fully double blind, simultaneous movement enabled

### Description

We're playing a large game semi-asynchronously (fully double blind, simultaneous movement), such that one player may be doing their movement turn when the other player is not actually connected.

When a player moves a unit past an opponent's hidden unit, the opponent is prompted to take an attack of opportunity against the unit being moved. If the opponent is not connected at the time, the server freezes, presumably since it is not able to send a message to the opponent's MM client - the dedicated server's logs say "Send Failed" when this happens.

When this occurs, the player is no longer able to perform any additional movement, and if either player attempts to re-connect to the server, their MM app sits loading it forever. The game must be restarted from the last autosave to be able to continue, and the player must re-do their movement, but the fact that there is a hidden unit nearby has been given away.

### Files

This is the game settings we use: 
[mjog-gameoptions.zip](https://github.com/MegaMek/megamek/files/6921371/mjog-gameoptions.zip)

**Comment by NickAragua** [comment]
This will require some thought on how to handle it. Currently, the server blocks and waits endlessly when expecting a "Client Feedback Request". 

Some approaches I've thought of include a timeout period after which some default value ("no" in the case of a hidden unit attack) is assumed by the server, or introducing a server command that signals the server that it's time to give it up. We'll have to discuss it internally for a bit.

**Comment by mjog** [comment]
If that's the issue, I wonder if the any pending Client Feedback Request packets could be cached by the server and re-sent to the appropriate client when the payer reconnects?

**Comment by mjog** [comment]
Would be happy to look into a fix for that if so.

**Comment by NickAragua** [comment]
It goes into an endless loop, but breaks out of the synchronized block every iteration (see Server.processPointblankShotCFR), so there may be an opportunity to detect a client re-connecting and try to resend it (you might have to somehow wait for the client to finish receiving the game update though).

**Comment by NickAragua** [comment]
BTW, @mjog are you on the megamek slack or megamek discord?

**Comment by mjog** [comment]
> BTW, @mjog are you on the megamek slack or megamek discord?

Nah, I'm a free software hippy, so don't use platform that those. ;)

You don't have a Matrix or Jabber server do you? If not, I'm happy to interact here.

**Comment by mjog** [comment]
I feel like the solution to this is maybe something similar to the solution I'd need to fix up Tactical Genius in #3070 - so will be looking for some commonality there.

Maybe there needs to be a "waiting for hidden unit" phase or something inserted so that infinite loop can be busted out of.

**Comment by mjog** [comment]
Thinking about this some more, aside from the networking issues I don't think there's a decent way of handling this without giving away the fact that there's a hidden unit in the location - the server can't notify other clients as to why it seems frozen without giving the hidden unit away, but also *not* notifying other clients gives away the fact that there's a hidden unit there because the player doing the move finds their unit frozen until the opposing player makes a decision about the taking the shot. They're dammed if they do, dammed if they don't.

It'd be unofficial, but I think the only way to handle this is property, especially for larger, async, double blind games, is to be able to give hidden units some orders when deploying them like "fire at will", "hold your fire", "fire on >= x to-hit", then let the server process those orders.

**Comment by HoneySkull** [comment]
What if you have the concept of a "null player" that accepts and responds with default actions when a player or bot is disconnected?
